paimon-codegen/src/main/java/org/apache/paimon/codegen/CodeGeneratorImpl.java

```diff
@@ -53,8 +53,10 @@ public GeneratedClass<RecordComparator> generateRecordComparator(
     }
 
     @Override
-    public GeneratedClass<RecordEqualiser> generateRecordEqualiser(List<DataType> fieldTypes) {
-        return new EqualiserCodeGenerator(RowType.builder().fields(fieldTypes).build())
+    public GeneratedClass<RecordEqualiser> generateRecordEqualiser(
+            List<DataType> fieldTypes, int[] ignoreFields) {
```

Comment:
What about use `ProjectedRow` in `LookupChangelogMergeFunctionWrapper`, So we not have to touch the codegen part.

Reply:
I tried using ProjectedRow, but it didn't seem to avoid changing the codegen section of the code.

Reply:
fix in https://github.com/apache/paimon/pull/4126/commits/31bdf2e272f485feefd159d5dd2de5cb0ae8f3dd